src/api/channels.test.tsx

```diff
@@ -286,24 +199,31 @@ describe('channels api functions', () => {
       });
 
       const expected: ChannelSummary = {
-        first_date: '2022-01-31T00:00:00',
+        first_date: '2022-01-29T00:00:00',
         most_recent_date: '2023-01-31T00:00:00',
-        recent_sample: [1, 2, 3],
+        recent_sample: [
+          { '2022-01-31T00:00:00': 6 },
+          { '2022-01-30T00:00:00': 5 },
+          { '2022-01-29T00:00:00': 4 },
+        ],
       };
-      expect(axios.get).toHaveBeenCalledWith(`/channels/summary/${'channel'}`, {
-        headers: { Authorization: 'Bearer null' },
-      });
+
       expect(result.current.data).toEqual(expected);
     });
 
-    it('does not send axios request to fetch channel summary when given no channel or system channel', async () => {
+    it('does not send request to fetch channel summary when given no channel or system channel', async () => {
+      let requestSent = false;
+      server.events.on('request:start', () => {
+        requestSent = true;
+      });
+
       const { result } = renderHook(() => useChannelSummary(undefined), {
         wrapper: hooksWrapperWithProviders(),
       });
 
       expect(result.current.isFetching).toBeFalsy();
       expect(result.current.isLoading).toBeTruthy();
-      expect(axios.get).not.toHaveBeenCalled();
+      expect(requestSent).toBe(false);
 
       const { result: result2 } = renderHook(
```

Comment:
Consider using the `rerender` function returned by `renderHook` here. The `renderHook` callback accepts arbitrary props as the first argument. New props can be passed to the hook through `rerender(newProps)`:

```ts
import { useChannelSummary } from './channels';
import { hooksWrapperWithProviders } from './setupTests';

const { rerender } = renderHook((channel?: string) => useChannelSummary(channel), {
  wrapper: hooksWrapperWithProviders(),
  // this will be forwarded to the `channel` param of the callback
  initialProps: undefined,
})

// ...

// re-render hook with another channel name.
rerender('timestamp');
```

Reply:
Ah, I didn't realise that there's a `rerender` function from `renderHook` - good catch this simplifies the test code significantly :)